src/foam/nanos/approval/ApprovableAwareDAO.js

```diff
@@ -0,0 +1,383 @@
+
+            approvalRequestDAO.put_(getX(), fulfilledRequest);
+
+            if ( fulfilledRequest.getStatus() == ApprovalStatus.APPROVED ) {
+              lifecycleObj.setLifecycleState(LifecycleState.ACTIVE);
+              return super.put_(x,obj);
+            } 
+
+            // create request has been rejected is only where we mark the object as REJECTED
+            lifecycleObj.setLifecycleState(LifecycleState.REJECTED);
+            return super.put_(x,obj); 
+          } 
+          if ( approvedObjCreateRequests.size() > 1 ) {
+            logger.error("Something went wrong cannot have multiple approved/rejected requests for the same request!");
+            throw new RuntimeException("Something went wrong cannot have multiple approved/rejected requests for the same request!");
+          } 
+
+          RoleApprovalRequest approvalRequest = new RoleApprovalRequest.Builder(getX())
+            .setDaoKey(getDaoKey())
+            .setObjId(approvableAwareObj.getApprovableKey())
+            .setClassification(getOf().getObjClass().getSimpleName())
+            .setOperation(Operations.CREATE)
+            .setInitiatingUser(((User) x.get("user")).getId())
+            .setStatus(ApprovalStatus.REQUESTED).build();
+
+          fullSend(getX(), approvalRequest, obj);
+
+          // we are storing the object in it's related dao with a lifecycle state of PENDING
+          UserFeedbackAware feedbackAwareObj = (UserFeedbackAware) obj;
+
+          UserFeedback newUserFeedback = new UserFeedback.Builder(getX())
+            .setStatus(UserFeedbackStatus.SUCCESS)
+            .setMessage("An approval request has been sent out.")
+            .setNext(feedbackAwareObj.getUserFeedback()).build();
+
+          FObject clonedObj = obj.fclone();
+
+          UserFeedbackAware feedbackAwareClonedObj = (UserFeedbackAware) clonedObj;
+
+          feedbackAwareClonedObj.setUserFeedback(newUserFeedback);
+
+          return super.put_(x,clonedObj);
+        } else {
+          logger.error("Something went wrong used an invalid lifecycle status for create!");
+          throw new RuntimeException("Something went wrong used an invalid lifecycle status for create!");
+        }
+      } else {
+        Iterator allProperties = obj.getClassInfo().getAxiomsByClass(PropertyInfo.class).iterator();
+
+        List<String> storageTransientPropertyNames = new ArrayList<>();
+
+        while ( allProperties.hasNext() ){
+          PropertyInfo prop = (PropertyInfo) allProperties.next();
+
+          if ( prop.getStorageTransient() ){
+            storageTransientPropertyNames.add(prop.getName());
+          }
+        }
+
+        // then handle the diff here and attach it into the approval request
+        Map updatedProperties = currentObjectInDAO.diff(obj);
+
+        for ( int i = 0; i < storageTransientPropertyNames.size(); i++ ){
+          updatedProperties.remove(storageTransientPropertyNames.get(i));
```

Comment:
why removing storageTransient?  They may trigger other DAO stack logic when the obj is finally 'put'.

Reply:
We're not removing it from the object thats getting passed, we're removing the storageTransientProperties from the diff object which was generated.